model/jpa/src/main/java/org/keycloak/organization/jpa/JpaOrganizationProvider.java

```diff
@@ -108,10 +162,39 @@ private OrganizationAdapter getAdapter(RealmModel realm, String id) {
         }
 
         if (!realm.getId().equals(entity.getRealmId())) {
-            return null;
+            throw new IllegalArgumentException("Organization [" + entity.getId() + " does not belong to realm [" + realm.getId() + "]");
+        }
+
+        return new OrganizationAdapter(entity, session);
+    }
+
+    private GroupModel createOrganizationGroup(RealmModel realm, String name) {
+        String groupName = getCanonicalGroupName(name);
+        GroupModel group = groupProvider.getGroupByName(realm, null, name);
+
+        if (group != null) {
+            throw new IllegalArgumentException("A group with the same already exist and it is bound to different organization");
```

Comment:
Not really, if the group exists with that name is because it is already bound to a organization.

Reply:
Probably it would be the case in most cases, but I guess there is nothing what prevents administrator to create or already have a regular group with any name. 

Do you plan to restrict it in a way that group with `kc.org.` prefix cannot exist unless it's an organization?

Reply:
Yes, the idea is to reserve such names exclusively to orgs.